lib/db.ts

```diff
@@ -465,3 +465,53 @@ export async function deleteFile(fileId: number) {
     .where(eq(Schemas.files1.id, fileId))
     .returning();
 }
+
+
+// OTP functions ==========================================================================================
+
+export async function saveOtp(phoneNumber: string, otp: string, expiresAt: string) {
```

Comment:
I'd like OTPs to be rate limited, I'm not sure what the industry standard is for rate limiting. I feel like it should be high enough not to be noticed by a human, but it should block bot nets from being able to spam the otp request. How do you suggest we do this? 

Also should a new OTP overwrite the old OTP?

Either way, to do this, select and insert should be done in a db.transaction((tx) => { ... })